File: NANDOR/NandorSettingsViewController.m
//
//  NandorSettingsViewController.m
//  NANDOR
//
//  Created by Chris Nandor on 2014.05.31.
//  Copyright (c) 2014 Pudge.Net. All rights reserved.
//

#import "NandorSettingsViewController.h"
#import "NANDOR-Swift.h"

@interface NandorSettingsViewController ()
@property (weak, nonatomic) IBOutlet UISwitch *quietSwitch;

@end

static NandorSingleton *_sharedData;

@implementation NandorSettingsViewController

- (void)loadView {
    [super loadView];

    if (self.sharedData.hornQuiet) {
        [self.quietSwitch setOn:true];
    }
}


- (IBAction)switchQuiet:(UISwitch *)sender {
    self.sharedData.hornQuiet = self.quietSwitch.isOn;
}

- (NandorSingleton *)sharedData {
    if (!_sharedData) _sharedData = [[NandorSingleton alloc] init];
    return _sharedData;
}



@end

File: NANDOR/View Controllers/NandorSettingsViewController.swift
//
//  NandorSettingsViewController.swift
//  NANDOR
//
//  Created by Chris Nandor on 2014.06.02.
//  Copyright (c) 2014 Pudge.Net. All rights reserved.
//

import UIKit

class NandorSettingsViewController:UIViewController {
    //@property (weak, nonatomic) IBOutlet UISwitch *quietSwitch;
    @IBOutlet var quietSwitch:UISwitch
    var singleton = NandorSingleton()
    
//    override func loadView() {
//        super.loadView()
//        self.quietSwitch.setOn(self.singleton.hornQuiet(), animated: true)
//    }

    @IBAction func switchQuiet(sender:UISwitch) {
        self.singleton.setHornQuiet(self.quietSwitch.state.getLogicValue())
    }
}
